src/v/kafka/client/topic_cache.cc

```diff
@@ -18,20 +18,30 @@ namespace kafka::client {
 
 ss::future<>
 topic_cache::apply(std::vector<metadata_response::topic>&& topics) {
-    leaders_t leaders;
```

Comment:
doing it here, because github doesn' tallow one to comment on the commit message. 

Can you write a comment on what the improvement *actually is* - otherwise is hard to tell if the intention is what the code says.